src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+            crossbleed_id: context.get_identifier("KNOB_OVHD_AIRCOND_XBLEED_Position".to_owned()),
+            ditching_id: context.get_identifier("OVHD_PRESS_DITCHING_PB_IS_ON".to_owned()),
+            engine_1_state_id: context.get_identifier("ENGINE_STATE:1".to_owned()),
+            engine_2_state_id: context.get_identifier("ENGINE_STATE:2".to_owned()),
+            eng_1_fire_id: context.get_identifier("Fire_ENG1_Agent1_Discharge".to_owned()),
+            eng_2_fire_id: context.get_identifier("Fire_ENG2_Agent1_Discharge".to_owned()),
+            ovhd_flow_selector_id: context
+                .get_identifier("KNOB_OVHD_AIRCOND_PACKFLOW_Position".to_owned()),
+
+            pack_flow_id: context.get_identifier("COND_PACK_FLOW".to_owned()),
+
+            flow_demand: Ratio::new::<percent>(0.),
+            absolute_flow: MassRate::new::<kilogram_per_second>(0.),
+            fcv_1_open_allowed: false,
+            fcv_2_open_allowed: false,
+            should_open_fcv: [false, false],
+
+            apu_bleed_on: false,
+            crossbleed_is_on: false,
+            ditching_is_on: false,
+            engine_1_in_start_mode: false,
+            engine_2_in_start_mode: false,
+            engine_1_on_fire: false,
+            engine_2_on_fire: false,
+            flow_control_valve_open: false,
+            flow_selector_position: OvhdFlowSelector::Norm,
+            pack_in_start_condition: false,
+            single_pack_operation: false,
+        }
+    }
+
+    fn update(
+        &mut self,
+        aircraft_state: &AcStateManager,
+        acs_overhead: &AirConditioningSystemOverhead,
+        engines: [&impl EngineCorrectedN1; 2],
+        pressurization: &impl CabinAltitude,
+        pack_flow_valve: &[PackFlowValve; 2],
+    ) {
+        self.flow_control_valve_open = pack_flow_valve.iter().any(|fcv| fcv.fcv_is_open());
+        self.single_pack_operation =
+            pack_flow_valve[0].fcv_is_open() != pack_flow_valve[1].fcv_is_open();
+        self.pack_in_start_condition = self.pack_start_condition_determination(pack_flow_valve);
+        // TODO: Add overheat protection
+        self.flow_demand = self.flow_demand_determination(aircraft_state);
+        self.absolute_flow = self.absolute_flow_calculation(pressurization);
+        self.fcv_open_allowed_determination(acs_overhead);
+        self.should_open_fcv = self.should_open_fcv_determination(engines);
+    }
+
+    fn pack_start_condition_determination(&mut self, pack_flow_valve: &[PackFlowValve; 2]) -> bool {
+        // Returns true when one of the packs is in start condition
+        pack_flow_valve
+            .iter()
+            .any(|fcv| fcv.fcv_timer() <= Duration::from_secs_f64(Self::PACK_START_TIME))
+    }
+
+    fn flow_demand_determination(&self, aircraft_state: &AcStateManager) -> Ratio {
+        let mut intermediate_flow =
+            Ratio::new::<percent>((self.flow_selector_position as u32) as f64);
```

Comment:
**suggestion**: Add a `From` implementation:

```rust
impl From<OvhdFlowSelector> for Ratio {
    fn from(value: OvhdFlowSelector) -> Self {
        Ratio::new::<percent>((value as u8) as f64)
    }
}
```
and then change this code to:
```suggestion
        let mut intermediate_flow: Ratio = self.flow_selector_position.into();
```

Reply:
Added! This is cool.